examples/async/completion.py

```diff
@@ -103,7 +103,7 @@ async def main() -> None:
         response = await s.execute()
 
         # print out all the options we got
-        for option in response.suggest.auto_complete[0].options:
+        for option in response.suggest["auto_complete"][0].options:
```

Comment:
To be more typing friendly I am changing all examples and tests where there is an attribute that is determined by a name given in the corresponding request to use dictionary syntax, since it is not possible to add type hints for variable names. The attribute format still works, but mypy/pyright does not like it.

Reply:
This is not super intuitive - also I don't understand "it is not possible to add type hints for variable names". Do you maybe have an example?

Reply:
Sorry, let's look at this specific code. The `auto_complete` attribute here is not an attribute of anything that is in the specification, it is just a name given in the request:

```python
s = s.suggest("auto_complete", text, completion={"field": "suggest"})
```

This is a common pattern in ES requests, it happens with many things, most prominent of all in aggregations, for which I'm also switched to use dictionary syntax. Does it make more sense now? How can we provide a type hint for `auto_complete` here, given that it is only known at runtime? But if we consider `suggest` to be a dictionary, then it is easier, because `auto_complete` is now a generic string key. Here is how pyright sees `suggest`:

![Screenshot 2024-10-18 at 2 32 35 PM](https://github.com/user-attachments/assets/8d2db00f-95a6-419a-a67e-91fe3f1c284a)

Reply:
I see, thank you 👍